### PR TITLE
doc api/overview: add overview document about when starting to use Groonga as library

### PR DESCRIPTION
**Pull request by cosmo0920**

### CODE REVIEW

File: doc/source/reference/api/overview.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Here is an example that uses Groonga as full-text search library.
+
+.. code-block :: c
+
+   grn_rc rc;
+   /* Preparing resource will be used by libgroonga. */
+   rc = grn_init();
+   if (rc != GRN_SUCCES) {
+     grn_fin();
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: grn_fin() isn't needed to be called on this case.

File: doc/source/reference/api/overview.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   /* Releasing resource used by libgroonga. */
+   grn_fin();
+   return 0;
+
+Reference
+---------
+
+.. c:function:: grn_rc grn_init(void)
+
+  `grn_init()` allocates resource that is used by libgroonga. It must call once before calling other Groonga API.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Use `initializes` instead of `allocates`.

File: doc/source/reference/api/overview.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Overview
+========
+
+Summary
+-------
+
+Groonga can use as library. You can start to use Groonga as full-text search
+library with the following API.
+
+``grn_init`` is initializer for libgroonga.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Use reference markup.

File: doc/source/reference/api/overview.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Here is an example that uses Groonga as full-text search library.
+
+.. code-block :: c
+
+   grn_rc rc;
+   /* Preparing resource will be used by libgroonga. */
+   rc = grn_init();
+   if (rc != GRN_SUCCESS) {
+     return 1;
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Use `EXIT_FAILURE` instead of `1`.

File: doc/source/reference/api/overview.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+   }
+   /* some Groonga API calling code... */
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Use `Some`.

Comment (kou) [inline doc/source/reference/api/overview.rst]: Use `codes`.

File: doc/source/reference/api/overview.rst [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+   /* Releasing resource used by libgroonga. */
+   grn_fin();
+   return 0;
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Use `EXIT_SUCCESS` instead of `0`.

File: doc/source/reference/api/overview.rst [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+Reference
+---------
+
+.. c:function:: grn_rc grn_init(void)
+
+  `grn_init()` initializes resource that is used by libgroonga. It must call once before calling other Groonga API.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: Wrong markup.

    ``grn_init()``

File: doc/source/reference/api/overview.rst [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+  :return: ``GRN_SUCCESS`` on success, not ``GRN_SUCCESS`` on error.
+
+.. c:function:: grn_rc grn_fin(void)
+
+  `grn_fin()` releases resource that is used by libgroonga. It must not call other Groonga API after calling `grn_fin()`.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: `It` -> `You`.

File: doc/source/reference/api/overview.rst (hunk 7 above)
Comment (kou) [inline doc/source/reference/api/overview.rst]: `It` -> `You`.
`it` is missing after `call`.

File: doc/source/reference/api/overview.rst [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+.. c:function:: grn_rc grn_init(void)
+
+  ``grn_init()`` initializes resource that is used by libgroonga. You must call it once before calling other Groonga API.
+
+  :return: ``GRN_SUCCESS`` on success, not ``GRN_SUCCESS`` on error.
+
+.. c:function:: grn_rc grn_fin(void)
+
+  ``grn_fin()`` releases resource that is used by libgroonga. You must not call it other Groonga API after calling :c:func:`grn_fin()`.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: `it` isn't needed.

Comment (kou) [inline doc/source/reference/api/overview.rst]: :c:func:\`grn_fin()\` isn't needed in `grn_fin()` document itself.

File: doc/source/reference/api/overview.rst [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+Overview
+========
+
+Summary
+-------
+
+Groonga can use as library. You can start to use Groonga as full-text search
+library with the following API.
+
+:c:func:`grn_init` is initializer for libgroonga.
```
Comment (kou) [inline doc/source/reference/api/overview.rst]: `()` will be needed after `grn_init`.

**Comment by kou** [comment]
Good work!

**Comment by cosmo0920** [comment]
Thank you for merging!

**Pull request merged**